spec/controllers/file_controller_spec.rb

```diff
@@ -12,6 +13,17 @@
   attr_reader :inv_file
   attr_reader :pathname
   attr_reader :basename
+  attr_reader :client
+
+  def mock_httpclient
```

Comment:
I think this approach is fine. For future reference take a look at WebMock which will block all external web traffic by default and will also give you handy errors (on how to stub it) when it catches any calls.

Reply:
Todo added above.